File: lib/policies.ts
import * as iam from "aws-cdk-lib/aws-iam";

export const cloudAdminPolicyDocument = new iam.PolicyDocument({
  statements: [
    new iam.PolicyStatement({
      actions: ["ec2:*", "s3:*", "iam:*", "rds:*", "cloudwatch:*"],
      resources: ["*"],
    }),
  ],
});

export const devOpsPolicyDocument = new iam.PolicyDocument({
  statements: [
    new iam.PolicyStatement({
      actions: [
        "codepipeline:*",
        "codebuild:*",
        "ecs:*",
        "cloudformation:*",
        "cloudwatch:*",
      ],
      resources: ["*"],
    }),
  ],
});

export const solutionsArchitectPolicyDocument = new iam.PolicyDocument({
  statements: [
    new iam.PolicyStatement({
      actions: ["ec2:*", "s3:*", "rds:*", "cloudfront:*", "lambda:*"],
      resources: ["*"],
    }),
  ],
});

export const securityEngineerPolicyDocument = new iam.PolicyDocument({
  statements: [
    new iam.PolicyStatement({
      actions: ["iam:*", "cloudtrail:*", "guardduty:*", "config:*", "kms:*"],
      resources: ["*"],
    }),
  ],
});

File: policies.py
from aws_cdk import aws_iam as iam

def cloud_admin_policy():
    return iam.PolicyStatement(
        actions=["ec2:*", "s3:*", "iam:*", "rds:*", "cloudwatch:*"],
        resources=["*"]
    )

def dev_ops_policy():
    return iam.PolicyStatement(
        actions=["codepipeline:*", "codebuild:*", "ecs:*", "cloudformation:*", "cloudwatch:*"],
        resources=["*"]
    )

def solutions_architect_policy():
    return iam.PolicyStatement(
        actions=["ec2:*", "s3:*", "rds:*", "cloudfront:*", "lambda:*"],
        resources=["*"]
    )

def security_engineer_policy():
    return iam.PolicyStatement(
        actions=["iam:*", "cloudtrail:*", "guardduty:*", "config:*", "kms:*"],
        resources=["*"]
    )